sources/constraint-metadata.asciidoc

```diff
@@ -382,8 +382,33 @@ This interface is returned by [methodname]`MethodDescriptor.getReturnValueDescri
 include::{validation-api-source-dir}javax/validation/metadata/ReturnValueDescriptor.java[lines=7..-1]
 ----
 
-[[constraintmetadata-constraintdescriptor]]
+[[constraintmetadata-containerdescriptor]]
+=== ContainerDescriptor and ContainerElementTypeDescriptor
+
+The `ContainerDescriptor` interface describes those elements that can host container element constraints (see <<constraintdeclarationvalidationprocess-containerelementconstraints>>).
+
+`ContainerDescriptor` is extended by `PropertyDescriptor`, `ParameterDescriptor`, `ReturnValueDescriptor` and `ContainerElementTypeDescriptor`.
+
+[source, JAVA, indent=0]
+----
+include::{validation-api-source-dir}javax/validation/metadata/ContainerDescriptor.java[lines=7..8;13..-1]
+----
+
+`getConstraintsForContainerElementType()` returns a descriptor representing the constraints of the specified type argument.
+`getConstrainedContainerElementTypes()` returns descriptors for all those type arguments that are hosting at least one constraint or are marked for cascaded validation.
```

Comment:
We had that discussion back in the 1.0 days, but I'm not sure whether we had a strong reason.